src/client/store/reducers/blocksReducer.js

```diff
@@ -395,9 +400,32 @@ const initialState = [
 ];
 
 export default function (state = initialState, action) {
-  const { blockName } = action;
+  const {
+    blockName,
+    jazzPresetArray,
+    rockPresetArray,
+    popPresetArray,
+    userPresetArray,
+  } = action;
 
   switch (action.type) {
+    case SET_DEFAULT_PRESET: {
+      return initialState;
+    }
+    case SET_JAZZ_PRESET: {
+      return jazzPresetArray;
+    }
+    case SET_ROCK_PRESET: {
+      return rockPresetArray;
+    }
+    case SET_POP_PRESET: {
+      return popPresetArray;
+    }
+    case SET_USER_PRESET: {
+      return state.map((effectFromStore, i) => (
```

Comment:
Reducers should be as simple as possible, move please this map to the action

Reply:
done)